test/testdata/desugar/sclass.rb.flatten-tree.exp

```diff
@@ -133,8 +133,10 @@ begin
 
     <emptyTree>
 
+    <emptyTree>
+
     def wrapper(<blk>)
-      <emptyTree>
+      :"e"
```

Comment:
@aisamanra can you speak to why this expectation file is changing?

Reply:
`def foo; end` in Ruby evaluates to the symbol `:foo`, so this replaces the moved methods with the symbols corresponding to the defined name.

As for why we need to, it's… kind of frustrating. Consider the example
```ruby
def foo
  private def bar; end
end
```
My initial instinct was that this should desugar to
```ruby
def foo; end
private(def bar; end)
```
but while this is fine for `private`, it's not fine for an arbitrary method `f`. Desugaring in that way would be a problem here:
```ruby
class C
  def f; end
  def foo
    f(def bar; end)
  end
end
```
If we desugared that as follows, we'd get an error about a missing method:
```ruby
class C
  def f; end
  def foo; end
  f(def bar; end)  # `self.f` does not exist on C
end
```
because `f` is not a static method. So instead, I desugar that to
```ruby
class C
  def f; end
  def foo; f(:bar); end
  def bar; end
end
```
which recaptures the semantics well enough and also gets the scoping right for `bar` and for `f`. I did so by making definitions always desugar to the relevant symbols: I can imagine also implementing a special case in which methods only desugar to their relevant symbols if they appear in a send, which would result in this diff being unchanged but keep the symbol around for places where it matters: it's not strictly necessary and it would complicate the logic a little bit but wouldn't be terribly hard, if we strongly feel like this should still desugar to `<emptyTree>` in this position.

Reply:
Thank you this makes a lot of sense.

Was this added as a test case somewhere? This would be a great test case.